### PR TITLE
Final stable havana fixes

### PR DESCRIPTION
**Pull request by retr0h**

### CODE REVIEW

File: nova/compute/manager.py [hunk 1]
```diff
@@ -4353,38 +4358,76 @@ def _heal_instance_info_cache(self, context):
...
+
+            self._instance_uuids_to_heal = instance_uuids
+        else:
+            # Find the next valid instance on the list
+            while instance_uuids:
                 try:
-                    instance = instance_obj.Instance.get_by_uuid(
-                        context, instance_uuids.pop(0),
-                        expected_attrs=['system_metadata', 'info_cache'])
+                    inst = instance_obj.Instance.get_by_uuid(
```
Comment (alop) [inline nova/compute/manager.py:4397]: Won't these two lines give an 'unexpected indent' error?

**Comment by alop** [comment]
Yes, this is a huge review. Many of the changes are around ESX, powervm, and xen. Some of the changes to the libvirt driver are carried over from our merge of the Inktank fork.
+1 from me if someone can confirm my comment above.

File: nova/compute/manager.py (hunk 1 above)
Comment (retr0h) [inline nova/compute/manager.py:4397]: Doesn't look like it.

**Comment by alop** [comment]
+1
LGTM, but someone else must approve.

**Comment by markvoelker** [comment]
+1.  Note that from the look of this we'll probably need to rebase a package spec patch or two as well.  I'll look into that when it hits the build gate.

**Pull request merged**